daml-lf/transaction/src/test/scala/com/digitalasset/daml/lf/transaction/TransactionSpec.scala

```diff
@@ -74,13 +74,14 @@ class TransactionSpec extends FreeSpec with Matchers with GeneratorDrivenPropert
         ),
         ImmArray(V.NodeId(0), V.NodeId(2)),
       )
-      val cids = tx.cids(
-        GenTransaction.cidConsumerInstance(
-          V.NodeId.cidConsumer,
-          V.ContractId.cidConsumer,
-          V.cidConsumerInstance(V.ContractId.cidConsumer))
-      )
-      cids shouldBe Set[V.ContractId]("cid0", "cid1", "cid2", dummyCid)
+
+      def collectCids(tx: Transaction): Set[V.ContractId] = {
```

Comment:
Minor: This test has become a bit pointless as it no longer tests the `cids` methods and it doesn't really test the `foreach` methods in full.

Reply:
I will keep it as example to show it is not dead code.

Reply:
by the way it is using the `cids` method.